manager/controllers/app/select_modules.go

```diff
@@ -430,3 +437,31 @@ func (p *PlotterGenerator) validateRestrictions(restrictions []adminrules.Restri
 	}
 	return true
 }
+
+func NestedFieldNoCopy(obj map[string]interface{}, fields ...string) (interface{}, bool, error) {
```

Comment:
Why did you implement this function on your own?

Reply:
Because the original method can only deal with nested maps, like map[a: map[b: map[c: d]]]. However, in our case, we should process something like map[capabilities: [ map[scope: workload], map[ a: b]]]. That's map>slice>map. So I add line 448-456 to address the slice part.